docs/README.md

```diff
@@ -0,0 +1,290 @@
+
+```typescript
+// Fetch the batches of a user.
+fetchUserBatches(address: string);
+
+// Fetch TCO2 token by it's address to get details like name, symbol, address and projectVintage.
+fetchTCO2TokenById(address: string);
+
+// Fetch TCO2Token with the symbol to get details like name, symbol, address and projectVintage. A symbol would look like this: `TCO2-VCS-<projectId>-YYYY`. Make sure to input the full symbol.
+fetchTCO2TokenByFullSymbol(symbol: string);
+
+// Fetch all TCO2s details like name, symbol, address and projectVintage.
+
+fetchAllTCO2Tokens();
+
+// Fetch all user retirements to get information like creationTX, amount, token details and certificate details.
+fetchUserRetirements(address: string, first?: number, skip?: number);
+
+// Fetch all redeems of a given pool to get information like amount, creator and token details. Only accepts the Pooltypes "NCT" or "BCT".
+fetchRedeems(poolSymbol: Pooltypes, first?: number, skip?: number);
+
+// Fetch all redeems of a user by pool and get information like amount and token details.
+fetchUserRedeems(address: string, poolSymbol: Pooltypes, first?: number, skip?: number);
+
+// Fetch TCO2 tokens that are part of the given pool and get information on name, amount, methodology and more.
+fetchPoolContents(poolSymbol: Pooltypes, first?: number, skip?: number);
+
+// Fetch a project by it's id to get more info on region and the standard.
+
+fetchProjectById();
+
+// Fetch all aggregations (including for example, tco2TotalRetired or totalCarbonBridged).
+fetchAggregations();
+```
+
+## Custom queries
+
+There's a lot more other pre-built subgraph queries that I could show you, but what I really want to show you is the `fetchCustomQuery` method.
+
+This allows you to fetch with your own queries and can be very powerful if you know graphQL. You can also check out a lot of example queries in our subgraph [playgrounds](https://thegraph.com/hosted-service/subgraph/toucanprotocol/matic).
+
+- Getting all infos on a project of a Carbon Credit
+
+```typescript
+import { gql } from "@urql/core";
+
+const query = gql`
+  query ($id: String) {
+    project(id: $id) {
+      projectId
+      region
+      standard
+      methodology
+      vintages {
+        id
+      }
+    }
+  }
+`;
+
+const result = await toucan.fetchCustomQuery(query, { id: "1" });
+```
+
+# AutoOffset using the OffsetHelper
+
+⚠️ This part is currently under construction and can't be used. ⚠️
+
+## AutoOffset CO2
+
+Using the OffsetHelper methods is the easiest way to offset CO2 right now as it handles the whole process for you.
+
+_(Uses Ubeswap/SushiSwap to acquire BCT/NCT, redeems it for TCO2s and retires them)_
+
+This is how you do that when you want to use ETH (or the network's native currency) in your balance.
+
+In this example, you'd be offsetting 1 TON of CO2 with carbon projects from the NCT pool.
+
+```typescript
+const offsetReceipt = await toucan.autoOffsetUsingETH("NCT", parseEther("1.0"));
```

Comment:
Just want to confirm if I understand the interface correctly. You're specifying how many tons of NCT you want and it will calculate the required ETH price for you?

Reply:
Yes, and thank you for flaggin this as the function name was also updated here to `autoOffsetExactOutETH`. You would use `autoOffsetExactOutETH` if you would want to offset e.g., 5 USDC